src/application/models/v2/User_model_v2.php

```diff
@@ -97,14 +97,29 @@ public function find_by_phone($phone)
         return $user;
     }
 
-    public function find_list_userId_by_fullName($fullName){
+    public function find_list_userId_by_fullName($fullName, $id_service_integrated){
         $idList = [];
         if($fullName == null || $fullName == ''){
             return $idList;
         }
         $ID_ROLES_OF_CUSTOMER = 3;
-        $sql = "SELECT id FROM ea_users WHERE CONCAT(first_name, ' ',last_name) LIKE ? AND id_roles = ?";
-        $result = $this->db->query($sql, array('%'.$fullName.'%', $ID_ROLES_OF_CUSTOMER))->result_array();
+
+        $result = [];
+        if(strlen($id_service_integrated) == 0){
+            $sql = "SELECT id FROM ea_users WHERE CONCAT(first_name, ' ',last_name) LIKE ? AND id_roles = ?";
+            $result = $this->db->query($sql, array('%'.$fullName.'%', $ID_ROLES_OF_CUSTOMER))->result_array();
+        }else{
+            $sql = "SELECT eaUsers.id 
+                    FROM ea_users eaUsers  
+                    WHERE CONCAT(eaUsers.first_name, ' ',eaUsers.last_name) LIKE ? 
+                          AND eaUsers.id_roles = ?
+                          AND eaUsers.id IN ( 
```

Comment:
id or id_user_integrated?

Reply:
it's id. 

ID in ea_users table is the foreign key in appointments table.

Reply:
ok, why do you make a condition: id IN set of eaAppointments.id_users_customer ?
this is false always ?

Reply:
Dạ cái luồng là như vậy  
Khi mình search = tên patient VÀ id_service_integrated, thì việc đầu tiên là 

Bước 1: mình sẽ vào ea_appointments để list ra hết những cái appontments liên quan tới id_service_integrated.

Bước 2: Minh query trong bảng EA_User, kiếm tất cả id mà có cái tên LIKE %paitent_name% mình gửi lên, do đây là phép toán LIKE nên nó sẽ trả về 1 list, ví dụ , patientName = Nguyen, thì sql sẽ trả về kết quả " Long Nguyen" , "Nguyen Canh" , "Suong Nguyen Thi Thu" , v.v....

Bước 3: Từ cái list id ea_user mình kiếm dc ở bước 2, mình so sanh với cái list_appointments ở bước 1 , để xem thằng nào vừa thỏa id_service_integrated, vừa có id_users_customer thõa với các params mình truyền trên url